src/App.tsx

```diff
@@ -1,39 +1,56 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+interface State {
+  clockName: string,
+  hasClock: boolean,
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId: number | null = null;
```

Comment:
You can give this a default value and keep it always as a number
```suggestion
  timerId = 0;
```